src/arrayMethodReduce.js

```diff
@@ -5,8 +5,20 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let prev = initialValue;
+    let fromIndex = 0;
+
+    if (arguments.length < 2) {
+      fromIndex = 1;
+      prev = this[0];
+    }
```

Comment:
It's a good case to use ternary operator to set correct prev and fromIndex values